src/Microsoft.CorrelationVector/CorrelationVectorV1.cs

```diff
@@ -0,0 +1,292 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Threading;
+using System.Globalization;
+
+namespace Microsoft.CorrelationVector
+{
+    /// <summary>
+    /// This class represents a lightweight vector for identifying and measuring
+    /// causality.
+    /// </summary>
+    public sealed class CorrelationVectorV1 : CorrelationVector
+    {
+        internal new const byte MaxVectorLength = 63;
+        internal const byte BaseLength = 16;
+
+        internal readonly string BaseVector = null;
+
+        private int extension = 0;
+
+        private bool immutable = false;
+
+        /// <summary>
+        /// This is the termination sign which should be used when vector length exceeds 
+        /// the <see cref="MaxVectorLength"/>
+        /// </summary>
+        public const string TerminationSign = "!";
+
+        /// <summary>
+        /// Creates a new correlation vector by extending an existing value. This should be
+        /// done at the entry point of an operation.
+        /// </summary>
+        /// <param name="correlationVector">
+        /// Taken from the message header indicated by <see cref="HeaderName"/>.
+        /// </param>
+        /// <returns>A new correlation vector extended from the current vector.</returns>
+        public new static CorrelationVectorV1 Extend(string correlationVector)
+        {
+            if (CorrelationVectorV1.IsImmutable(correlationVector))
+            {
+                return CorrelationVectorV1.Parse(correlationVector);
+            }
+
+            if (CorrelationVectorV1.ValidateCorrelationVectorDuringCreation)
+            {
+                CorrelationVectorV1.Validate(correlationVector);
+            }
+
+            if (CorrelationVectorV1.IsOversized(correlationVector, 0))
+            {
+                return CorrelationVectorV1.Parse(correlationVector + CorrelationVectorV1.TerminationSign);
+            }
+
+            return new CorrelationVectorV1(correlationVector, 0, false);
+        }
+
+        /// <summary>
+        /// Not supported in V1.
+        /// </summary>
+        /// <param name="correlationVector">
+        /// Taken from the message header indicated by <see cref="HeaderName"/>.
+        /// </param>
+        /// <returns>A new correlation vector extended from the current vector.</returns>
+        public new static CorrelationVectorV1 Spin(string correlationVector)
```

Comment:
We don't need this method since it is not called by the CorrelationVector class.

Reply:
This is here in case someone tries to directly call CorrelationVectorV1.Spin

Reply:
I dont think we need to entertain that since calling CorrelationVectorV1.Spin will anyways call CorrelationVector.Spin if you dont have this method